core/src/com/fourinachamber/fortyfive/screen/general/CustomActors.kt

```diff
@@ -1196,61 +1213,224 @@ open class CustomHorizontalGroup(
     override var isHoveredOver: Boolean = false
     override var isClicked: Boolean = false
 
+    var forcedPrefWidth: Float? = null
+    var forcedPrefHeight: Float? = null
+
+    private val onLayout: MutableList<() -> Unit> = mutableListOf()
+
     private val backgroundHandleObserver = SubscribeableObserver<String?>(null)
     override var backgroundHandle: String? by backgroundHandleObserver
 
     private val background: Drawable? by automaticResourceGetter<Drawable>(backgroundHandleObserver, screen)
 
-
     override fun draw(batch: Batch?, parentAlpha: Float) {
-        val (x, y) = localToStageCoordinates(Vector2(0f, 0f))
         background?.draw(batch, x, y, width, height)
         super.draw(batch, parentAlpha)
     }
 
+    override fun onLayout(callback: () -> Unit) {
+        onLayout.add(callback)
+    }
+
+    override fun layout() {
+        onLayout.forEach { it() }
+        // These cant use normal iterators because .validate may use an iterator over children as well,
+        // and LibGDX doesn't like that
+        (0 until children.size).forEach { (children[it] as? Layout)?.validate() }
+        super.layout()
+    }
+
+    fun invalidateChildren() {
+        children.forEach { (it as? Layout)?.invalidate() }
+    }
+
     override fun resortZIndices() {
         children.sort { el1, el2 ->
             (if (el1 is ZIndexActor) el1.fixedZIndex else -1) -
                     (if (el2 is ZIndexActor) el2.fixedZIndex else -1)
         }
     }
 
+    override fun getPrefWidth(): Float = forcedPrefWidth ?: super.getPrefWidth()
+    override fun getPrefHeight(): Float = forcedPrefHeight ?: super.getPrefHeight()
+
     override fun initStyles(screen: OnjScreen) {
         addActorStyles(screen)
     }
 }
 
+open class CustomGroup(
+    override val screen: OnjScreen
+) : WidgetGroup(), ZIndexGroup, ZIndexActor, BackgroundActor, HasOnjScreen, OffSettable, OnLayoutActor {
+
+    override var offsetX: Float = 0f
+    override var offsetY: Float = 0f
+
+    override var fixedZIndex: Int = 0
+
+    var forcedPrefWidth: Float? = null
+    var forcedPrefHeight: Float? = null
+
+    private val onLayout: MutableList<() -> Unit> = mutableListOf()
+
+    private val backgroundHandleObserver = SubscribeableObserver<String?>(null)
+    override var backgroundHandle: String? by backgroundHandleObserver
+
+    private val background: Drawable? by automaticResourceGetter<Drawable>(backgroundHandleObserver, screen)
+
+    override fun draw(batch: Batch?, parentAlpha: Float) {
+        this.x += offsetX
+        this.y += offsetY
+        background?.draw(batch, x, y, width, height)
+        super.draw(batch, parentAlpha)
+        this.x -= offsetX
+        this.y -= offsetY
+    }
+
+    override fun onLayout(callback: () -> Unit) {
+        onLayout.add(callback)
+    }
+
+    override fun layout() {
+        onLayout.forEach { it() }
+        (0 until children.size).forEach { (children[it] as? Layout)?.validate() }
```

Comment:
why not: 
`children.filterIsInstance<Layout>().forEach { it.validate() }`
or
`children.forEach { (it as? Layout)?.validate() }`

you have this 3 times in the code